service/matching/config.go

```diff
@@ -147,7 +151,7 @@ func NewConfig(dc *dynamicconfig.Collection) *Config {
 		ShutdownDrainDuration:                 dc.GetDurationProperty(dynamicconfig.MatchingShutdownDrainDuration, 0*time.Second),
 		VersionCompatibleSetLimitPerQueue:     dc.GetIntProperty(dynamicconfig.VersionCompatibleSetLimitPerQueue, 10),
 		VersionBuildIdLimitPerQueue:           dc.GetIntProperty(dynamicconfig.VersionBuildIdLimitPerQueue, 1000),
-		UserDataPollFrequency:                 dc.GetDurationProperty(dynamicconfig.MatchingUserDataPollFrequency, 5*time.Minute),
+		GetUserDataLongPollTimeout:            dc.GetDurationProperty(dynamicconfig.MatchingGetUserDataLongPollTimeout, 5*time.Minute),
```

Comment:
I hope this is a good default, is should be long enough if the cluster is active and the nodes are communicating regularly.
Otherwise, it should also be fine AFAICT, because by the time this request times out we'll have unloaded the task queue.
In any case, connection loss will eventually be detected and the caller will retry.

Reply:
Yeah, I'm not totally sure but here was my reasoning:
If nodes go up and down "normally" the clients should get connection errors and retry. The case where a server goes away and the client doesn't realize until the timeout should be rare (network partition, dropped packets, etc.) Of course it can happen, but 5m is probably fine in that case